src/EventListener/LegacyRequestValueSynchronizingListener.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2016 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\EventListener;
+
+use Contao\CoreBundle\Request\ValueAdapter;
+use Contao\Environment;
+use Contao\Input;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * Sets the current request on the \Input and \Environment classes.
+ *
+ * @author Christian Schiffler <https://github.com/discordier>
+ *
+ * @deprecated Deprecated since Contao 4.3, to be removed in Contao 5.0.
+ *             Use the request or request stack instead.
+ */
+class LegacyRequestValueSynchronizingListener implements EventSubscriberInterface
```

Comment:
Changed in 80c0dfe, however, I do not remember that we made such a decision.
Have we documented somewhere that we do not want to use subscribers?